sycl/include/CL/sycl/queue.hpp

```diff
@@ -22,6 +22,39 @@
 
 #include <utility>
 
+// having _TWO_ mid-param #ifdefs makes the functions very difficult to read.
+// Here we simplify the &CodeLoc declaration to be _CODELOCPARAM(&CodeLoc) and
+// _CODELOCARG(&CodeLoc) Similarly, the KernelFunc param is simplified to be
+// _KERNELFUNCPARAM(KernelFunc) Once the queue kernel functions are defined,
+// these macros are #undef immediately.
+
+// replace _CODELOCPARAM(&CodeLoc) with nothing
+// or :   , const detail::code_location &CodeLoc =
+// detail::code_location::current()
+// replace _CODELOCARG(&CodeLoc) with nothing
+// or :  const detail::code_location &CodeLoc = {}
+
+#ifndef DISABLE_SYCL_INSTRUMENTATION_METADATA
+#define _CODELOCPARAM(a)                                                       \
+  , const detail::code_location a = detail::code_location::current()
+
+#define _CODELOCARG(a)
+#define _CODELOCFW(a) , a
```

Comment:
This must be undefined as well.
See https://github.com/intel/llvm/blob/a42d4a4d99bb800f25a0d70e8c84be4fa991584f/sycl/include/CL/sycl/queue.hpp#L711-L714

Reply:
Oh, good catch.
Done.